src/frontend/App/hooks/useHentOppgaverForOpprettelse.ts

```diff
@@ -0,0 +1,59 @@
+import { RessursFeilet, RessursStatus, RessursSuksess } from '../typer/ressurs';
+import { useApp } from '../context/AppContext';
+import { useCallback, useState } from 'react';
+import {
+    IOppgaverForOpprettelse,
+    OppgaveForOpprettelseType,
+} from '../../Komponenter/Behandling/Totrinnskontroll/OppgaverForOpprettelse';
+
+export interface HentOppgaverForOpprettelseState {
+    feilmelding: string | undefined;
+    hentOppgaverForOpprettelse: (behandlingId: string) => void;
+    oppgaverForOpprettelse: IOppgaverForOpprettelse;
+    oppdaterOppgavetyperSomSkalOpprettes: (typer: OppgaveForOpprettelseType[]) => void;
+}
+
+const initialState: IOppgaverForOpprettelse = {
+    oppgavetyperSomSkalOpprettes: [],
+    oppgavetyperSomKanOpprettes: [],
+};
+
+export const useHentOppgaverForOpprettelse = (): HentOppgaverForOpprettelseState => {
+    const { axiosRequest } = useApp();
+    const [oppgaverForOpprettelse, settOppgaverForOpprettelse] =
+        useState<IOppgaverForOpprettelse>(initialState);
+    const [feilmelding, settFeilmelding] = useState<string>();
+
+    const hentOppgaverForOpprettelse = useCallback(
+        (behandlingId: string) => {
+            settFeilmelding(undefined);
+            axiosRequest<IOppgaverForOpprettelse, undefined>({
+                method: 'GET',
+                url: `/familie-ef-sak/api/oppgaverforopprettelse/${behandlingId}`,
+            }).then((res: RessursSuksess<IOppgaverForOpprettelse> | RessursFeilet) => {
+                if (res.status === RessursStatus.SUKSESS) {
+                    settOppgaverForOpprettelse(res.data);
+                } else {
+                    settFeilmelding(res.frontendFeilmelding);
+                }
+            });
+        },
+        [axiosRequest]
+    );
```

Comment:
Syntes "Oppgaver for opprettelse" er litt forvirrende. Kunne man hatt `oppgaverSomKanOpprettes` og `oppgaverSomSkalOpprettes` adskilt? 

Og returneres begge her? Eller er det kun oppgaver som _kan_ opprettes som egentlig hentes?

Reply:
Ja, litt vrient å komme på noe mere lesbart ifm "kan" og "skal" opprette. Litt av tanken her er å hente OppgaverForOpprettelse som har både både kanOpprette og skalOpprette.